src/main/java/io/kestra/plugin/airflow/dags/TriggerDagRun.java

```diff
@@ -0,0 +1,218 @@
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Slf4j
+@Schema(
+    title = "Trigger Airflow DAG",
+    description = "Trigger an Airflow DAG run and wait for its completion."
+)
+@Plugin(
+    examples = {
+        @Example(
+            title = "Basic authorization",
+            code = {
+                "  - id: trigger_dag",
+                "    type: io.kestra.plugin.airflow.TriggerDagRun",
+                "    baseUrl: http://airflow.example.com",
+                "    dagId: example_dag",
+                "    checkFrequency: PT30S",
+                "      interval: PT30S",
+                "      maxIterations: 100",
+                "      maxDuration: PT1H",
+                "    options:",
+                "      basicAuthUser: myusername",
+                "      basicAuthPassword: mypassword"
+            }
+        ),
+        @Example(
+            title = "Bearer authorization",
+            code = {
+                "  - id: trigger_dag",
+                "    type: io.kestra.plugin.airflow.TriggerDagRun",
+                "    baseUrl: http://airflow.example.com",
+                "    dagId: example_dag",
+                "    checkFrequency: PT30S",
+                "      interval: PT30S",
+                "    headers:",
+                "      authorization: 'Bearer {{ TOKEN }}'"
+            }
+        ),
+        @Example(
+            title = "Basic authorization. Custom body",
+            code = {
+                "  - id: trigger_dag",
+                "    type: io.kestra.plugin.airflow.TriggerDagRun",
+                "    baseUrl: http://airflow.example.com",
+                "    dagId: example_dag",
+                "    checkFrequency: PT30S",
+                "      interval: PT30S",
+                "    options:",
+                "      basicAuthUser: myusername",
+                "      basicAuthPassword: mypassword",
+                "    body: |",
+                "       {",
+                "         \"conf\": {",
+                "           \"source\": \"kestra\",",
+                "           \"flow\": \"{{ flow.id }}\",",
+                "           \"namespace\": \"{{ flow.namespace }}\",",
+                "           \"task\": \"{{ task.id }}\",",
+                "           \"execution\": \"{{ execution.id }}\"",
+                "           }",
+                "       }"
+            }
+        ),
+    }
+)
+public class TriggerDagRun extends AirflowConnection implements RunnableTask<TriggerDagRun.Output> {
+
+    @Schema(
+        title = "The ID of the DAG to trigger"
+    )
+    @NotNull
+    @PluginProperty(dynamic = true)
+    private String dagId;
+
+    @Schema(
+        title = "The frequency at which to check the DAG run status"
+    )
+    @PluginProperty
+    @Builder.Default
+    private WaitFor.CheckFrequency checkFrequency = WaitFor.CheckFrequency.builder().build();
```

Comment:
I don't think it's a good idea to use another task internal property class from the core plugin in another plugin.
I think here using only a timeout and ping the DAG endpoint each second would be enough it's what we did in a lot of other tasks.

Reply:
I didn't see the description from Anna.

You can keep the options but I'd rather not use CheckFrequency and include a basic polling as we do in other plugins with only a maxDuration and a pollFrequency, you can take as example: https://github.com/kestra-io/plugin-airbyte/blob/master/src/main/java/io/kestra/plugin/airbyte/connections/CheckStatus.java